ttc.c

```diff
@@ -181,14 +186,63 @@ void sendString()
 	txChar("Hello World");
 }
 
-void sensorTest() {
-	int x = adc_read(0);
-	if (x > 0x00ff) {
-		txChar("High");
+int getTemp() { //returnt de temperatuur in tienden van graden C
+	float temp = tempSensor();
+	// Adafruit over de TMP36:
+	// Temp in C = (input(mv) - 500) / 10
+	temp = (((temp * 5 / 1024) - 0.5) * 100); // bereken temperatuur
+	int tempC = (temp * 10);
+	return(tempC);
+}
+
+void checkRx() { //checkt of er een bericht is binnengekomen op rx en schrijft het naar een variabele
+	if (UCSR0A && RXC0) {
+		//er is een bericht ontvangen
+		int firstInt = (UDR0 * 0x100); //schrijft het bericht naar de bovenste helft van een int
+		firstInt += rx(); //alle berichten bestaan uit 16 bits, hier word de tweede helft geschreven
+		if (firstInt == 0xffff) { //0xffff betekend dat het het begin is van een bericht is, de rest van het bericht wordt nu naar een variabele geschreven
+			uint32_t message = (rx() * 0x1000000);
+			message += (rx() * 0x10000);
+			message += (rx() * 0x100);
+			message += rx();
+			lastMessage = message;
+			handleRx();
+		}
+	}
+}
+
+handleRx() {
+	//leest lastMessage en neemt de bijbehorede acties
+	int command = (lastMessage % 0x1000); //het commando (bovenste 16 bits)
+	int payload = (lastMessage - command); //de payload van het bericht
+
+	switch(command) {
+		case 11:
+			tempOn = payload;
+			break;
+		case 12:
+			tempOff = payload;
+			break;
+		case 51:
+			setScreen(0x255);
```

Comment:
Waarschijnlijk bedoel je dit?
```suggestion
			setScreen(0xff);
```

Reply:
Ja, dat klopt wel ongeveer